packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -252,7 +252,14 @@ class PageController extends ScrollController {
   void attach(ScrollPosition position) {
     super.attach(position);
     final _PagePosition pagePosition = position;
-    pagePosition.viewportFraction = viewportFraction;
+
+    if (pagePosition.viewportFraction == viewportFraction)
+      return;
+    final double oldPage = pagePosition.page;
+    pagePosition._viewportFraction = viewportFraction;
+    if (oldPage != null) {
+      pagePosition.correctPixels(pagePosition.getPixelsFromPage(oldPage));
```

Comment:
according to documentation of correctPixels, this function should only be called during layout phase
can you add an assert something like
assert(SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks, 'attach can only be called during SchedulerPhase.persistentCallbacks')

If this assumption no longer held, would be better to update the documentation of correctPixels to include this case

Reply:
should really only be called when debugDoingLayout is true for the viewport in question